### PR TITLE
Integration with Solday's ERC6551 Implementation

### LINKED ISSUE #83: Use Solady's ERC6551 implementation
- We have to inherit from ERC6551 and override `_isValidSigner()` to use our AccessController
- If we use the upgradeable version, it will also use isValidSigner(). We have to be SUPER careful with AccessController in this context, we may have to parse if the call is part of an upgrade from msg.data

### PR DESCRIPTION
**Pull request by kingster-will**
#### Overview
This PR introduces integration with Solday's ERC6551 smart contract. Key modifications include the update of the `state` variable, renaming it from `nonce` and changing its type from `uint256` to `hash (bytes32)`. This adjustment ensures that `state` is now deterministic, computed from the input parameters of the `execute()` function.

#### Modifications
- **Inheritance:** IPAccount now extends Solday's ERC6551 contract to utilize its implementation of ERC6551 interfaces including token(), owner(), receiver functions and 1271 signature validation.
- **State/nonce Update:** Changed the type of `state` from `uint256` to `bytes32` according to Solady 6551, which is hash computed based on `execute()` inputs. All execute functions `execute()', `executeBatch()` and `executeWithSig()` share the function to update 'state' which calculate hash based on common inputs of execute functions: `to`, `value` and `data`.
- **Method Overrides:** Overrode `execute()` and `executeBatch()` methods from Solady ERC6551 to include check permissions via AccessController. This ensures that every execution undergoes a permission verification process.

## Test Plan 
All new added code covered by automation tests.

## Related Issue
Closes #83

### CODE REVIEW

File: contracts/IPAccountImpl.sol [hunk 1]
```diff
@@ -56,42 +55,30 @@ contract IPAccountImpl is IPAccountStorage, IIPAccount {
...
+        bytes calldata data
+    ) public view override(ERC6551, IIPAccount) returns (bytes4) {
+        return super.isValidSigner(signer, data);
     }
 
     /// @notice Returns the owner of the IP Account.
     /// @return The address of the owner.
-    function owner() public view returns (address) {
-        (uint256 chainId, address contractAddress, uint256 tokenId) = token();
-        if (chainId != block.chainid) return address(0);
```
Comment (LeoHChen) [inline contracts/IPAccountImpl.sol:93]: we don't need to do chainId checking?

**Review by LeoHChen — APPROVED** [review]
LGTM, needs @vectorized's review

File: test/foundry/access/AccessController.t.sol [hunk 2]
```diff
@@ -1372,12 +1372,14 @@ contract AccessControllerTest is BaseTest {
         );
         vm.prank(owner);
         mockNFT.burn(tokenId);
-        vm.expectRevert(abi.encodeWithSelector(ERC721NonexistentToken.selector, tokenId));
-        accessController.getPermission(
-            address(ipAccount),
-            signer,
-            address(mockModule),
-            mockModule.executeSuccessfully.selector
+        assertEq(
```
Comment (jdubpark) [inline test/foundry/access/AccessController.t.sol:1375]: Why is the expectRevert removed & replaced with assertEq? Behavior change?

File: test/foundry/access/AccessController.t.sol [hunk 3]
```diff
@@ -1406,7 +1408,15 @@ contract AccessControllerTest is BaseTest {
         vm.prank(owner);
         mockNFT.burn(tokenId);
 
-        vm.expectRevert(abi.encodeWithSelector(ERC721NonexistentToken.selector, tokenId));
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Errors.AccessController__PermissionDenied.selector,
```
Comment (jdubpark) [inline test/foundry/access/AccessController.t.sol:1413]: Same here, change in error behavior?

**Comment by jdubpark** [comment]
For tests, it'd be better to use parameterized internal functions for repetitive code, e.g. `newState` and signatures, to improve readability. Pretty hard to scroll through the test cases without getting confused.

File: contracts/IPAccountImpl.sol (hunk 1 above)
Comment (kingster-will) [reply contracts/IPAccountImpl.sol:93]: Yes, we need this check. 
This check has been covered in Solady 6551's owner() function. as IPAccount inherit from Solady 6551 now.
we delegate the owner() to solady at Line 76  ` return super.owner();`

File: test/foundry/access/AccessController.t.sol (hunk 2 above)
Comment (kingster-will) [reply test/foundry/access/AccessController.t.sol:1375]: correct, the behavior is changing as migrate to solady, the soladay 6551 owner() doesn't revert when bound NFT was burnt, it return zero address instead.

File: test/foundry/access/AccessController.t.sol (hunk 3 above)
Comment (kingster-will) [reply test/foundry/access/AccessController.t.sol:1413]: yes, same reason above.

**Review by Vectorized — APPROVED** [review]
LGTM

**Review by Ramarti — APPROVED** [review]
Looks good

**Pull request merged**